logging-sample/README.adoc

```diff
@@ -3,59 +3,99 @@
+When opening the ui for the first time you need to login to Graylog (user: admin - passwort: admin).
 
-TODO: Define GELF Input in Graylog
+Graylog supports different sources of log messages.
+Before it is able to receive any message you need to tell it where to listen.
 
+1. Visit to http://localhost:9000[http://localhost:9000]
+1. Login to Graylog (user: admin - passwort: admin)
+1. In the top menu select "System" > "Inputs"
+1. Select "GELF UDP" as the input type and click "Launch new input".
+1. In the popup select the "global" checkbox and give the input a title (i.e. GELF UDP)
+1. Click "save".
++
 image::.readme/graylog-gelf-input.png[]
-
-Once Graylog is started the 2 Spring Boot application that are part of the sample can be started.
-Go to the `server-spring-log4j` or `server-spring-logback` subfolder and use maven to start the server:
-`mvn spring-boot:run`
+1. Click "Show received messages" to the right of the newly added input.
+1. Click on the little play symbol in the top right corner to enable auto refresh of the page.
+1. Launch the server-spring-log4j or the server-spring-logback example (`mvn spring-boot:run`).
+1. Click on a message to see all the details.
++
+image::.readme/graylog-logging.png[]
 
 The server application will send the complete logging to the Graylog server by using the GELF format.
-In Graylog you can inspect the application logging and see all the Rico context based metadata that is part of each logging message.
+You can see the complete Rico context based metadata that is part of each logging message in the Graylog UI.
 
-image::.readme/graylog-logging.png[]
 
 The server applications both provide a http endpoint that can be called to create additional logging.
 To call the endpoint just open http://localhost:8080/api/sample[http://localhost:8080/api/sample].
 
 === Using the ELK stack
 
-TODO: documentation for ELK stack
+The sample provides docker containers to easily start a preconfigured ELK stack that is used to store the log messages.
+The complete setup can be found in the `docker-elk` folder.
+The `docker-compose` tool is needed to run the sample.
+If `docker-compose` is installed on your system you only need to execute the `start.sh` from `docker-elk` folder to boot the ELK stack.
+Once this is done the web ui of Kibana can be accessed at http://localhost:5601[http://localhost:5601].
+
+INFO::
+If the ELK stack fails to start with the message `max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]` or similar than have a look at https://github.com/docker-library/elasticsearch/issues/111[elasticsearch issue #111]
+
+=== Configuring Kibana
+
+The ELK stack is ready to receive log messages without a configuration.
+In order to browse the messages a little configuration is needed:
+
+1. Launch the server-spring-log4j or the server-spring-logback example (`mvn spring-boot:run`).
+1. Visit http://localhost:5601[http://localhost:5601].
```

Comment:
same as above: 1,2,3...

Reply:
no, this automatically renders to a correctly numbered list, no need to count by hand ;-)

Reply:
see https://github.com/rico-projects/rico-samples/blob/42174900b067fbf46ea6991d7622c26fa34787a5/logging-sample/README.adoc ;-)